src/analytics/navigationTracker/navigationTracker.ts

```diff
@@ -0,0 +1,118 @@
+import { BaseTracker } from "../baseTracker/baseTracker";
+import { NavigationEventInterface } from "./navigationTracker.interface";
+import { validateParameter } from "../../utils/validateParameter";
+
+export class NavigationTracker extends BaseTracker {
+  eventName: string = "event_data";
+
+  constructor() {
+    super();
+    this.initialiseEventListener();
+  }
+
+  /**
+   * Initialises the event listener for the document click event.
+   *
+   * @param {type} None
+   * @return {type} None
+   */
+  initialiseEventListener() {
+    document.addEventListener("click", this.trackNavigation.bind(this), false);
+  }
+
+  /**
+   * Tracks the page load event and sends the relevant data to the data layer.
+   *
+   * @param {PageViewParametersInterface} parameters - The parameters for the page view event.
+   * @return {boolean} Returns true if the event was successfully tracked, false otherwise.
+   */
+  trackNavigation(event: Event): boolean {
+    const element: HTMLLinkElement = event.target as HTMLLinkElement;
+    /**
+     * Navigation tracker is only for links and buttons
+     */
+    if (element.tagName !== "A" && element.tagName !== "BUTTON") {
```

Comment:
same for here

Reply:
When I take event.target (the page element which was affected by the click) and assign to my variable element, I use the type HTMLLinkElement.
element become an object (with all attributes and method) and I don't know why (I suppose it's because of the type) tagName contains the tag in CAPS. So I have to check with CAPS tag in my condition.